Makefile

```diff
@@ -165,10 +165,18 @@ taskmetadata-validator:
 ecr-execution-role-image:
 	$(MAKE) -C misc/ecr $(MFLAGS)
 
+gocyclo:
+    # Run gocyclo over all .go files in the agent, excluding vendor/ and testutils/ directories, and all *_test.go and *_mocks.go files
+	gocyclo -over 15 $(shell go list -f "{{$p := .}}{{range $f := .GoFiles}}{{$p.Dir}}/{{$f}} {{end}}" ./agent/... | grep -v /vendor/ | grep -v /testutils/ | grep -v _test\.go$ | grep -v _mocks\.go$)
```

Comment:
why 15 and not lower?

Reply:
goreportcard reports ones over 15. I can make it lower but lot more work needs to be done to refactor more code.